### PR TITLE
applications: add authentification with  RegistryConfigFile in HelmCredentials

### PR DESCRIPTION
**Pull request by vgramer**
**What does this PR do / Why do we need it**:
Thsi PR just update the crd and validation, logic will be implemented in next PR

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/validation/applicationdefinition.go [hunk 1]
```diff
@@ -90,14 +90,33 @@ func validateSource(source appskubermaticv1.ApplicationSource, f *field.Path) []
...
 }
 
+func validateHelmCredentials(credential *appskubermaticv1.HelmCredentials, f *field.Path) *field.Error {
+	if credential != nil {
+		if credential.RegistryConfigFile != nil && (credential.Username != nil || credential.Password != nil) {
+			return field.Forbidden(f.Child("registryConfigFile"), "registryConfigFile can not be used in conjunction with username / password")
+		}
+
+		if credential.Username != nil && credential.Password == nil {
+			return field.Forbidden(f.Child("password"), "password must be definedif username if defined")
```
Comment (ahmedwaleedmalik) [inline pkg/validation/applicationdefinition.go]:
```suggestion
			return field.Forbidden(f.Child("password"), "password must be specified with username")
```

File: pkg/validation/applicationdefinition.go [hunk 2]
```diff
@@ -90,14 +90,33 @@ func validateSource(source appskubermaticv1.ApplicationSource, f *field.Path) []
...
+		}
+		if credential.Password != nil && credential.Username == nil {
+			return field.Forbidden(f.Child("username"), "username must be definedif password if defined")
```
Comment (ahmedwaleedmalik) [inline pkg/validation/applicationdefinition.go]:
```suggestion
			return field.Forbidden(f.Child("username"), "username must be specified  with password")
```

**Review by ahmedwaleedmalik — COMMENTED** [review]
nits

File: pkg/validation/applicationdefinition.go (hunk 1 above)
Comment (vgramer) [reply pkg/validation/applicationdefinition.go]: much better :)

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by xrstf** [comment]
/hold

temporarily

**Comment by xrstf** [comment]
/hold cancel
/retest

Thanks for waiting.

**Pull request merged**